### DISCUSSION TITLE
Why use both mini.indentscope and indent-blankline?

### DISCUSSION THREAD

**Discussion by andradei** [LazyVim/LazyVim#362] [Q&A] [answered]
I'm new to customizing neovim, lua, and this codebase.
Why are both `echasnovski/mini.indentscope` and `lukas-reineke/indent-blankline` in this repo?
[mini.indentscope's documentation](https://github.com/echasnovski/mini.indentscope/blob/main/doc/mini-indentscope.txt#L43) show this:

# Comparisons

- 'lukas-reineke/indent-blankline.nvim':
    - Its main functionality is about showing static guides of indent levels.
    - Implementation of 'mini.indentscope' is similar to
      'indent-blankline.nvim' (using |extmarks| on first column to be shown
      even on blank lines). They can be used simultaneously, but it will
      lead to one of the visualizations being on top (hiding) of another.

---

What is the each of these plugins doing that are unique among each other? Maybe `mini.indentline` does the job of both.

**Comment by folke** [comment, marked as answer]
Both mini.indentscope and indent-blankline.nvim are useful for visualizing and operating on indent scope. mini.indentscope is an active indent guide that provides text objects and motions to operate on scope, while indent-blankline.nvim is a static guide that shows indent levels. They can be used simultaneously, but one will be on top of the other, hiding it.

You can disable any of the two if you want and see for yourself what would be different

**Reply by andradei** [reply]
Did just that, now it makes sense that the indent line animation of `echasnovski/mini.indentscope` appears on top of the indent line provided by `lukas-reineke/indent-blankline`, thus one complement the other.

Thank you.
